### PR TITLE
Add wallet address generator binary

### PR DESCRIPTION
**Pull request by OBorce**
- used for generating new seed phrase and addresses for mainnet

### CODE REVIEW

File: wallet/wallet-address-generator-lib/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+#[derive(Parser, Debug)]
+#[clap(version)]
+pub struct CliArgs {
+    /// Which network to generate the addresses for
+    #[arg(long, value_enum, default_value_t = Network::Mainnet)]
+    pub network: Network,
+
+    /// Number of addresses to generate and display
+    #[clap(long, default_value_t = 20)]
```
Comment (TheQuantumPhysicist) [inline wallet/wallet-address-generator-lib/src/lib.rs:59]: I think the default should be `1`.

File: wallet/wallet-address-generator-lib/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    pub number_addresses: u8,
```
Comment (TheQuantumPhysicist) [inline wallet/wallet-address-generator-lib/src/lib.rs:60]: address_count

File: wallet/wallet-address-generator-lib/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    InvalidInput(String),
+    #[error("Invalid mnemonic: {0}")]
+    InvalidMnemonic(wallet_controller::mnemonic::Error),
+    #[error("WalletError error: {0}")]
+    WalletError(#[from] WalletError),
+}
+
+pub fn run(args: CliArgs) -> Result<(), CliError> {
+    ensure!(
+        args.number_addresses <= 20,
```
Comment (TheQuantumPhysicist) [inline wallet/wallet-address-generator-lib/src/lib.rs:79]: This number should come from the default gap in the wallet, I'd say. Is it possible to do that?

File: wallet/wallet-address-generator-lib/src/lib.rs (hunk 2 above)
Comment (TheQuantumPhysicist) [inline wallet/wallet-address-generator-lib/src/lib.rs:60]: address_count

File: wallet/wallet-address-generator-lib/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            Network::Regtest => ChainType::Regtest,
+            Network::Signet => ChainType::Signet,
+        }
+    }
+}
+
+#[derive(Parser, Debug)]
+#[clap(version)]
+pub struct CliArgs {
+    /// Which network to generate the addresses for
```
Comment (TheQuantumPhysicist) [inline wallet/wallet-address-generator-lib/src/lib.rs:54]: The network, for which addresses will be generated

File: wallet/wallet-address-generator-lib/src/lib.rs (hunk 3 above)
Comment (OBorce) [reply wallet/wallet-address-generator-lib/src/lib.rs]: Fixed used the LOOKAHEAD_SIZE constant from the keychain

**Review by TheQuantumPhysicist — APPROVED** [review]

**Pull request merged**